File: Deepseek-Coder-33b-Instruct/stage2/declare_directive.f90
program main
  use openacc
  implicit none
  integer, parameter :: N = 100
  integer :: a(N), i, errors
  logical :: failed

  ! Initialize the array
  do i = 1, N
    a(i) = i
  end do

  !$acc data copyin(a) copyout(a)
  !$acc declare present(a)
  !$acc parallel loop
  do i = 1, N
    a(i) = a(i) * 2
  end do
  !$acc end parallel loop
  !$acc end data

  ! Check the results
  errors = 0
  do i = 1, N
    if (a(i) /= i*2) then
      errors = errors + 1
    end if
  end do

  if (errors == 0) then
    print *, "Test passed."
  else
    print *, "Test failed with ", errors, " errors."
  end if

end program main

File: Deepseek-Coder-33b-Instruct/stage2/declare_directive.cpp
#include <iostream>
#include <stdlib.h>

#pragma acc declare copy(a)
int a;

int main() {
    a = 5;

    #pragma acc parallel copy(a)
    {
        a = a * 2;
    }

    std::cout << "Value of a: " << a << std::endl;

    return 0;
}